Uses AppID for multi-user installation on IKS (#2327)
kubeflow/website: content/en/docs/ibm/deploy/install-kubeflow.md

* **${KF_DIR}** - The full path to your Kubeflow application directory.

### Multi-user, auth-enabled

Run the following steps to deploy Kubeflow with [IBM Cloud AppID](https://cloud.ibm.com/catalog/services/app-id)
as an authentication provider. 

The scenario is a Kubeflow cluster admin configures Kubeflow as a web
application in AppID and manages user authentication with builtin identity
providers (Cloud Directory, SAML, social log-in with Google or Facebook etc.) or
custom providers.

1. Follow the guide [Getting started with App ID](https://cloud.ibm.com/docs/appid?topic=appid-getting-started)
to create an AppID service instance.
2. Follow the step [Registering your app](https://cloud.ibm.com/docs/appid?topic=appid-app#app-register)

> to create an application with type _regularwebapp_ under the provisoned AppID

Author: Thanks, fixed inhttps://github.com/kubeflow/website/pull/2327/commits/a6ffbf3b299e35eba3302e5dd4bee73a7a9ee295

* **${KF_DIR}** - The full path to your Kubeflow application directory.

### Multi-user, auth-enabled

Run the following steps to deploy Kubeflow with [IBM Cloud AppID](https://cloud.ibm.com/catalog/services/app-id)

> as authentication provider. 

Reviewer: ```suggestion
as an authentication provider. 
```

> The scenario is a Kubeflow cluster admin configures Kubeflow as an web

Reviewer: ```suggestion
The scenario is a Kubeflow cluster admin configures Kubeflow as a web
```

application in AppID and manages user authentication with builtin

> identity providers (Cloud Directory, SAML, etc) or custom providers.

Reviewer: ```suggestion
identity providers (Cloud Directory, SAML, social log-in with Google or Facebook etc.) or custom providers.
```

1. Follow the guide [Getting started with App ID](https://cloud.ibm.com/docs/appid?topic=appid-getting-started)
to create an AppID service instance.

> 2. Follow the step [Registering your app](https://cloud.ibm.com/docs/appid?topic=appid-app#app-register)

Reviewer 2: 2. Follow the step [Registering your app](https://cloud.ibm.com/docs/appid?topic=appid-app#app-register)
to create an application with type _regularwebapp_ under the provisioned AppID

> 2. Follow the step [Registering your app](https://cloud.ibm.com/docs/appid?topic=appid-app#app-register)

Reviewer 2: ```
2. Follow the step [Registering your app](https://cloud.ibm.com/docs/appid?topic=appid-app#app-register)
to create an application with type _reguarwebapp_ under the provioned AppID
```

> 2. Follow the step [Registering your app](https://cloud.ibm.com/docs/appid?topic=appid-app#app-register)
> to create an application with type _reguarwebapp_ under the provioned AppID

Reviewer 2: ```suggestion
2. Follow the step [Registering your app](https://cloud.ibm.com/docs/appid?topic=appid-app#app-register)
to create an application with type _regularwebapp_ under the provisioned AppID
```

instance. Make sure the _scope_ contains _email_. Then retrieve the following 
configuration parameters from your AppID:
    * `clientId`
    * `secret`
    * `oAuthServerUrl`
3. Follow the step [Adding redirect URIs](https://cloud.ibm.com/docs/appid?topic=appid-managing-idp)

> to fill a URL for AppID to redirect to Kubeflow. the URL should look like `https://<kubeflow-FQDN>/login/oidc`. Notice that you could follow the guide [Securing the Kubeflow authentication with HTTPS](../authentication/) for the value of `<kubeflow-FQDN>`.

Reviewer 2: ```suggestion
to fill a URL for AppID to redirect to Kubeflow. the URL should look like `https://<kubeflow-FQDN>/login/oidc`. Notice that you could follow the guide [Securing the Kubeflow authentication with HTTPS](./authentication/) for the value of `<kubeflow-FQDN>`.
```

> to fill a URL for AppID to redirect to Kubeflow. the URL should look like `https://<kubeflow-FQDN>/login/oidc`. Notice that you could follow the guide [Securing the Kubeflow authentication with HTTPS](../authentication/) for the value of `<kubeflow-FQDN>`.

Reviewer 2: This seem like a circular thing. 

We need the deployment to be complete for to have a istio-ingress ip that gives us the FQDN and to complete the deployment we need the FQDN. 

There needs to be a bit more clarity.

Author: Moved this step to the very end after kubeflow deployment. see https://github.com/kubeflow/website/pull/2327/commits/a15edf41b62c5fbce380029f8fff7cfe9ad54943
